### DISCUSSION TITLE
Is there any way you can update the source code?

### DISCUSSION THREAD

**Discussion by astrolemonade** [nickcoutsos/keymap-editor#133] [General]
Hey!
First of all, I really appreciate and respect you for making this!
I recently got myself into the split keyboard world and build myself a choc Corne that happens to use ZMK. When a friend of mine sent me the link to the keymap editor I was very happy since the ZMK ecosystem is pretty sparse.

I read the statement from the Readme and also the one from the wiki: https://github.com/nickcoutsos/keymap-editor/wiki/Source-Code-Updates . I am really sorry you got eaten alive by greedy companies that want free stuff and I hope you are ok.
Is there any way you can update the source code? Maybe change the license to a more restrictive one like AGPL that most companies fear of: https://choosealicense.com/licenses/agpl-3.0 or just make it only source available and nobody should make commercial versions of it or maybe add ways to sponsor you.
I really hope that this didn't upset you!

**Comment by nickcoutsos** [comment]
Hi, thank you for your interest!

I don't think licensing would make the difference here. The distinction is one of selling the software versus making the software available and using that to provide value for their product.

The short answer is that I'm no longer holding out for this to become a community project. With that being the case it doesn't matter to me if forks are open source or not.

Outside of testing I actually have very little use for this editor, but I enjoy working on it and I think accepting donations would probably complicate that.

> I really hope that this didn't upset you!

Thanks for saying so, it's a difficult subject for me and I've had a lot of different feelings throughout the whole process to say the least.

**Comment by astrolemonade** [comment]
> The distinction is one of selling the software versus making the software available and using that to provide value for their product.

I really think this frustration was experienced by the VIA developers as well and there is nothing wrong in feeling that way. 

I used to have keyboards with proprietary configurator+-firmware but now I don't even buy a keyboard if it doesn't have opensource firmware.
I am seeing lots of keyboards that have a selling point the fact that they are configurable via QMK/VIA and I doubt any money from goes to VIA or QMK team.  Maybe they can assist you in finding what's good for you(there is a Discord server where you can get in touch with the devs linked on their website https://www.caniusevia.com , they are very friendly).

> The short answer is that I'm no longer holding out for this to become a community project. With that being the case it doesn't matter to me if forks are open source or not.

This project becoming a community project would be the next major milestone for ZMK ecosystem as a whole. At the end of the day, your feelings come first in making this decision, do what's best for your soul.

**Reply by nickcoutsos** [reply]
> This project becoming a community project would be the next major milestone for ZMK ecosystem as a whole.

I would guess at best that there might be a brief period between ZMK having a protocol for BLE-based keymap updates and getting ZMK Studio in a position where it's usable. Even then, there's never been much interaction so I'm still stumbling along on my own trying to find solutions to managing ZMK keymap data.

**Comment by BenBlumer** [comment]
Hey Nick,

Thanks for your work on this. It's such a cool tool. I'm new to custom keyboards. I just put together a wireless chocofi. I'm now learning the differences between zmk and qmk. One of the major things that seems to be missing from ZMK is the ecosystem (things like the configurator) and your work is really great! 

It's your software. You're awesome for sharing what you have. You owe us nothing. It would also be awesome to see the source code updated -- even under a customized license that hits hard at what bugs you. Maybe "You can't rehost for a for-profit project. If you want support for your custom keyboard on my site, pay up". If it's something you're interested in, and you are comfortable talking about what behaviours you're looking to curb, I'd be happy to help with customizing a license.

Regardless of how you move forward, thank you for this tool!
